### PR TITLE
HydroMechanics non-equilibrium initial states.

### PR DESCRIPTION
**Pull request by endJunction**
Adding same non-equilibrium initial states as for the SmallDeformation process #https://github.com/ufz/ogs/pull/2501.

Meanwhile the theoretical ideas were developed further and this is only for the comparison with the next new implementation and also provides the project files/tests. The new idea is to include the non-equilibrium in the residuum.

1. [x] Feature description was added to the [changelog](https://github.com/ufz/ogs/wiki/Release-notes-6.2.1)
2. [x] Tests covering your feature were added?
3. [x] Any new feature or behavior change was documented?

### CODE REVIEW

File: Documentation/ProjectFile/prj/processes/process/HYDRO_MECHANICS/nonequilibrium_state_variables/i_nonequilibrium_state_variables.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Non-equilibrium initial state variables which are not explicitly in equilibrium
```
Comment (wenqing) [inline Documentation/ProjectFile/prj/processes/process/HYDRO_MECHANICS/nonequilibrium_state_variables/i_nonequilibrium_state_variables.md]: not explicitly --> implicitly.

**Comment by wenqing** [comment]
Will the changes in the local assembler be in another PR?

**Comment by endJunction** [comment]
Yes, the F^{oob} (out of balance forces) formulation will follow in other PR.

File: ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp [hunk 2]
```diff
@@ -188,13 +188,45 @@ std::unique_ptr<Process> createHydroMechanicsProcess(
         config.getConfigParameter<double>(
             "reference_temperature", std::numeric_limits<double>::quiet_NaN());
 
+    // Non-equilibrium variables
+    ParameterLib::Parameter<double> const* nonequilibrium_stress = nullptr;
```
Comment (TomFischer) [inline ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp]: Just out of curiosity: Why not ´unique_ptr`?

File: ProcessLib/HydroMechanics/HydroMechanicsProcessData.h [hunk 3]
```diff
@@ -97,6 +101,12 @@ struct HydroMechanicsProcessData
     ParameterLib::Parameter<double> const& porosity;
     /// Solid's density. A scalar quantity, ParameterLib::Parameter<double>.
     ParameterLib::Parameter<double> const& solid_density;
+
+    /// Optional, non-equilibrium stress parameter.
+    ParameterLib::Parameter<double> const* const nonequilibrium_stress;
```
Comment (TomFischer) [inline ProcessLib/HydroMechanics/HydroMechanicsProcessData.h]: Why not ´unique_ptr`?

File: Tests/Data/HydroMechanics/InitialStates/displacement/into_initial_state_pcs_0.pvd [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
+<VTKFile type="Collection" version="0.1" byte_order="LittleEndian" compressor="vtkZLibDataCompressor">
+  <Collection>
+    <DataSet timestep="0" group="" part="0" file="into_initial_state_pcs_0_ts_0_t_0.000000.vtu"/>
+  </Collection>
+</VTKFile>
```
Comment (TomFischer) [inline Tests/Data/HydroMechanics/InitialStates/displacement/into_initial_state_pcs_0.pvd]: Is the pvd file needed?

**Comment by TomFischer** [comment]
Are the gml files needed? I didn´t see references in the prooject files.

File: ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp (hunk 2 above)
Comment (endJunction) [reply ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp]: It's a non-owning optional parameter.

**Comment by wenqing** [comment]
I guess that Tests/Data/HydroMechanics/InitialStates/pressure/Docu_HM_Non_Equilibrium_Initial_States.pdf  is for the documentation of one benchmark. However the web content file, which should refer to that PDF is missing.

File: ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h [hunk 5]
```diff
@@ -269,11 +302,11 @@ void HydroMechanicsLocalAssembler<ShapeFunctionDisplacement,
 
     // pressure equation
     local_rhs.template segment<pressure_size>(pressure_index).noalias() -=
-        laplace_p * p + storage_p * p_dot + Kpu * u_dot;
+        laplace_p * (p - p_neq) + storage_p * p_dot + Kpu * u_dot;
```
Comment (wenqing) [inline ProcessLib/HydroMechanics/HydroMechanicsFEM-impl.h]: I think that every simulation starts from the previous status, whether which is equilibrium (may from the previous simulation) or non-equilibrium (specfied). Therefore p_neq should be p0, and local_rhs should not be changed.  This is a suggestion for future change. The PR itself is OK with the current consideration.

**Comment by nagelt** [comment]
This should be no longer needed with the CompensateInitial....

**Comment by endJunction** [comment]
That's true, but I still didn't redo the test cases...

**Pull request closed without merging**